lib/litmus_paper/cache.rb

```diff
@@ -0,0 +1,33 @@
+require 'yaml'
+
+module LitmusPaper
+  class Cache
+    def initialize(location, namespace, ttl)
+      @path = File.join(location, namespace)
+      @ttl = ttl
+
+      FileUtils.mkdir_p(@path)
+    end
+
+    def set(key, value)
+      return unless @ttl > 0
+      File.open(File.join(@path, key), "a") do |f|
+        f.flock(File::LOCK_EX)
```

Comment:
What happens if we get two processes trying to update the cache simultaneously?

Reply:
As I understand it, flock is built for synchronizing file I/O across processes. This call will block until all shared flocks (`f.flock(File::LOCK_SH)`) exit and then acquire the exclusive lock (`f.flock(File::LOCK_EX)`) which will block all readers from reading the file until the lock is released.

Reply:
I would suggest using
```
f.flock(File::LOCK_EX | File::LOCK_NB)
```
and if locking fails, just run the usual health check

Reply:
@lollipopman I'm not convinced that's the best solution. There is likely to be brief, periodic lock contention when two processes attempt to set the cache at the same time. when this is the case, it will likely be faster to wait for the lock to be released and then read the cached value instead of making an external request. 

Thoughts?

Reply:
yeah, I do worry about a process hanging, and holding the lock, but I think the approach is a reasonable start